pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -88,14 +88,29 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./fix-qemu-ga.patch
-    ./9p-ignore-noatime.patch
     # Cocoa clipboard support only works on macOS 10.14+
     (fetchpatch {
       url = "https://gitlab.com/qemu-project/qemu/-/commit/7e3e20d89129614f4a7b2451fe321cc6ccca3b76.diff";
       sha256 = "09xz06g57wxbacic617pq9c0qb7nly42gif0raplldn5lw964xl2";
       revert = true;
     })
-  ] ++ lib.optional nixosTestRunner ./force-uid0-on-9p.patch;
+  ] ++ lib.optional stdenv.isLinux ./9p-ignore-noatime.patch
+    ++ lib.optional stdenv.isDarwin [
+    # 9p-darwin for 7.0 backported to 6.2.0
+    # Can generally be removed when updating derivation to 7.0, with the
+    # following caveat:
+    #
+    # QEMU upstream does not demand compatibility to pre-10.13, so 9p-darwin
+    # support on nixrequires utimensat fallback. The final commit in this patch
+    # set (f787ff6c128da93c2fd2ae92add20617a9941b48) was removed during the
+    # process of upstreaming this functionality, and will still be needed in nix
+    # until the macOS SDK reaches 10.13+.
+    (fetchpatch {
+      url = "https://github.com/willcohen/qemu/compare/v6.2.0..f787ff6c128da93c2fd2ae92add20617a9941b48.diff";
```

Comment:
Can we fetch this from the QEMU 9p.next tree instead? Just because it's a little more official. https://github.com/cschoenebeck/qemu/commits/9p.next

Reply:
Not entirely. Most of the patches will apply cleanly, but `meson.build` in the final commit conflicts when you go all the way back to 6.2.0. I know that 9p.next has other changes brewing, so depending on how that pull is eventually organized, a few other commits MAY break. Either way, the conflict commits will need to be modified, plus the utimensat fallback.

To get it as clean as possible, I suppose we could pull all but those last two commits from 9p.next, and then just have these final ones come from my fork. I'm also happy to just have the upstream commits be upstream, and then we check in these last two as patch files here, but based on some perusal of other PRs it seems like (and correct me if I'm wrong!) but fetchpatch is preferred to filling up this repo with extra checked-in files.

Reply:
> I'm also happy to just have the upstream commits be upstream, and then we check in these last two as patch files here

Let's do this.  fetchpatch is preferred when fetching from upstream, but I think it makes more sense (for longevity if nothing else) to include Nixpkgs-specific patches in Nixpkgs than in a git tree of whoever happened to submit it.

Reply:
I might let this sit a day or two longer if that's okay. There was one small change as the pull request from the 9p maintainer was being review. I suspect it'll be applied to the true upstream tree rather soon and that'll be the cleanest change longevity-wise.